src/main/java/de/tum/cit/aet/artemis/buildagent/dto/BuildJobQueueItem.java

```diff
@@ -14,8 +14,8 @@
 // in the future are migrated or cleared. Changes should be communicated in release notes as potentially breaking changes.
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public record BuildJobQueueItem(String id, String name, String buildAgentAddress, long participationId, long courseId, long exerciseId, int retryCount, int priority,
-        BuildStatus status, RepositoryInfo repositoryInfo, JobTimingInfo jobTimingInfo, BuildConfig buildConfig, ResultDTO submissionResult) implements Serializable {
+public record BuildJobQueueItem(String id, String name, String buildAgentName, String buildAgentAddress, long participationId, long courseId, long exerciseId, int retryCount,
```

Comment:
I suggest to create a `BuildAgent` record and store `shortName`, `displayName` and `address` in it. This allows better reuse and avoid too long parameter lists